### PR TITLE
feat: added wrappers around near-cli to support saving cmd to history

### PR DESCRIPTION
**Pull request by frolvanya**
This PR introduces shell-specific functions for Bash, Zsh, and Fish shells to improve NEAR CLI's integration with shell history. The provided functions ensure that commands executed via the near CLI are automatically added to the shell's history, making them accessible through arrow keys and standard history mechanisms

### CODE REVIEW

File: README.md
Comment (dj8yfo) [inline README.md:142]: this doc section can be linked to in a separate md file. 
this usually improves readability, as there's less of a shock on a reader, 
as he/she doesn't undergo reading all of the details at once, but sees document structure first, 
and can narrow his/her context and attention on a specific section later

Comment (frolvanya) [reply README.md]: Yep, makes sense. I was thinking about that, but decided just to put everything at the bottom for some reason. Fixed in 9b5687087718b885b0b986c80e80089a54a77652

**Review by frol — DISMISSED** [review]
@race-of-sloths score 8

**Comment by race-of-sloths** [comment]
@frolvanya Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Do you want to apply for monthly streak? Get 8+ score for a single PR this month and receive boost for race-of-sloths!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/frolvanya?type=bot&pr=near/near-cli-rs/415&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/frolvanya?type=bot&pr=near/near-cli-rs/415&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/frolvanya)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/frolvanya)

For maintainers:
- Score pull requests that participate in the Race of Sloths and receive a reward
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: docs/SHELL_HISTORY_INTEGRATION.md
Comment (frolvanya) [reply docs/SHELL_HISTORY_INTEGRATION.md:51]: Fixed in https://github.com/near/near-cli-rs/pull/415/commits/b08d6eab5c5c17e2c112d6e5fc82508f67ccfed3

**Review by frol — APPROVED** [review]

**Pull request merged**
